src/storage/txn/scheduler.rs

```diff
@@ -786,6 +794,17 @@ impl<E: Engine, L: LockManager> Scheduler<E, L> {
         };
         SCHED_STAGE_COUNTER_VEC.get(tag).write.inc();
 
+        let throttle = self
+            .inner
+            .quota_limiter
+            .consume_write(write_bytes, cost_time);
```

Comment:
if the result is a error, it's early return above, then the cost time is not recorded here

Reply:
@hunterlxt I think we can solve this by implementing `Drop` for the sample in https://github.com/tikv/tikv/pull/12151#discussion_r827676224, in which the duration will be consumed from limiter (but no delay is triggered).

Reply:
ok i'll move this block earlier